purchase_work_acceptance/models/work_acceptance.py

```diff
@@ -0,0 +1,218 @@
+    )
+    date_receive = fields.Datetime(
+        string='Received Date',
+        default=fields.Datetime.now,
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    date_accept = fields.Datetime(
+        string='Accepted Date',
+        readonly=True,
+    )
+    invoice_ref = fields.Char(
+        string='Invoice Reference',
+        copy=False,
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Vendor',
+        required=True,
+        change_default=True,
+        track_visibility='always',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    responsible_id = fields.Many2one(
+        comodel_name='hr.employee',
+        string='Responsible Person',
+        required=True,
+        change_default=True,
+        track_visibility='always',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+        string='Currency',
+        default=lambda self: self.env.user.company_id.currency_id.id,
+        required=True,
+        readonly=True,
+    )
+    state = fields.Selection(
+        [('draft', 'Draft'),
+         ('accept', 'Accepted'),
+         ('cancel', 'Cancelled')],
+        string='Status',
+        readonly=True,
+        index=True,
+        copy=False,
+        default='draft',
+        track_visibility='onchange'
+    )
+    wa_line_ids = fields.One2many(
+        comodel_name='work.acceptance.line',
+        inverse_name='wa_id',
+        string='Work Acceptance Lines',
+    )
+    notes = fields.Text(
+        string='Notes',
+    )
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        related='wa_line_ids.product_id',
+        string='Product',
+        readonly=False,
+    )
+    user_id = fields.Many2one(
+        comodel_name='res.users',
+        string='Work Acceptance Representative',
+        default=lambda self: self.env.user,
+        index=True,
+        track_visibility='onchange',
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        default=lambda self: self.env.user.company_id.id,
+        required=True,
+        index=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    purchase_id = fields.Many2one(
+        comodel_name='purchase.order',
+        string='Purchase Order',
+        readonly=True,
+    )
+
+    @api.model
+    def create(self, vals):
+        if vals.get('name', 'New') == 'New':
+            vals['name'] = self.env['ir.sequence'].next_by_code(
+                'work.acceptance') or '/'
+        return super(WorkAcceptance, self).create(vals)
+
+    @api.multi
+    def button_accept(self, force=False):
+        self._unlink_zero_quantity()
+        self.write({'state': 'accept', 'date_accept': fields.Datetime.now()})
```

Comment:
Here, it would be great to set the received_qty of the lines of services :smile:

Reply:
or no consu, product

Reply:
Edited : https://github.com/OCA/purchase-workflow/pull/774/commits/74b8c5f27338d714143dfee5372105bf91e269cc